x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/results/TrainedModelDefinitionChunk.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.dataframe.process.results;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig;
+import org.elasticsearch.xpack.ml.inference.persistence.TrainedModelDefinitionDoc;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class TrainedModelDefinitionChunk implements ToXContentObject {
+
+    private static final ParseField DEFINITION = new ParseField("definition");
+    private static final ParseField DOC_NUM = new ParseField("doc_num");
+    private static final ParseField EOS = new ParseField("eos");
+
+    public static final ConstructingObjectParser<TrainedModelDefinitionChunk, Void> PARSER = new ConstructingObjectParser<>(
+        "chunked_trained_model_definition",
+        a -> new TrainedModelDefinitionChunk((String) a[0], (Integer) a[1], (Boolean) a[2]));
+
+    static {
+        PARSER.declareString(constructorArg(), DEFINITION);
+        PARSER.declareInt(constructorArg(), DOC_NUM);
+        PARSER.declareBoolean(optionalConstructorArg(), EOS);
+    }
+
+    private final String definition;
+    private final int docNum;
+    private final Boolean eos;
```

Comment:
do we need a 3rd state (`null`)?
in code it looks like `null` and `false` are `false`.

it seems simpler to me to use `boolean` and handle `null` as part of parsing